docs/trace/customizing-the-sdk/README.md

```diff
@@ -359,10 +359,143 @@ using OpenTelemetry;
 Sdk.SetDefaultTextMapPropagator(new MyCustomPropagator());
 ```
 
-## Dependency Injection Support
+## Dependency injection support
+
+### Overview
+
+**Note** This information applies to the OpenTelemetry SDK version 1.4.0 and
+newer only.
+
+The SDK implementation of `TracerProviderBuilder` is backed by an
+`IServiceCollection` and supports a wide range of APIs to enable what is
+generally known as [dependency
+injection](https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection).
+
+For the below examples imagine an exporter with this constructor:
+
+```csharp
+    public class MyCustomExporter : BaseExporter<Activity>
+    {
+        public MyCustomExporter(
+            ILogger<MyCustomExporter> logger,
+            MyCustomService myCustomService)
+        {
+            // Implementation not important
+        }
+    }
+```
+
+We want to inject the `ILogger<MyCustomExporter>` and `MyCustomService`
+dependencies into our `MyCustomExporter` instance.
+
+#### Using Sdk.CreateTracerProviderBuilder()
+
+To register `MyCustomExporter` and `MyCustomService` we can use the
+`ConfigureServices` and `AddExporter` methods:
+
+```csharp
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .ConfigureServices(services =>
+    {
+        services.AddLogging();
+        services.AddSingleton<MyCustomService>();
+    })
+    .AddExporter<MyCustomExporter>(ExportProcessorType.Batch)
+    .Build();
+```
+
+When using the `Sdk.CreateTracerProviderBuilder` method the `TracerProvider`
+owns its own `IServiceCollection`. It will only be able to see services
+registered into that collection.
+
+#### Using the [OpenTelemetry.Extensions.Hosting](../../../src/OpenTelemetry.Extensions.Hosting/README.md) package
+
+**Note** If you are authoring an ASP.NET Core application or using the [.NET
+Generic Host](https://learn.microsoft.com/dotnet/core/extensions/generic-host)
+this is the recommended mechanism.
+
+```csharp
+builder.Services.AddSingleton<MyCustomService>();
```

Comment:
possible confusion with this build and the otel builder.. lets use a different name?

Reply:
Updated